### ISSUE TITLE
<Modal />: missing accessibility features

### ISSUE THREAD

**Issue by Radek-Palisa** [kiwicom/orbit-components#435]
## Expected Behavior
I am not sure if this is a bug or feature request, but it's so essential that I think it should be classified as bug and be given priority.

In a modal I expect this behaviour that is currently missing:

1) On modal close, the focus will return to the element which has opened the modal.
2) On modal open, the screenreader should announce the title of the modal, so `aria-label` or `aria-labelledby` needs to be included. 
3) On modal open, the focus should be placed on the first focusable element in the modal or on the close button. Since this is not happening at the moment, esc keydown doesn't trigger modal close until I click somewhere inside the modal.
4) While modal is open, the focus should be trapped within the modal's content. This means that if the current activeElement is that last focusable element inside the modal, pressing the tab key will move me to the first focusable element inside the modal and so the focus will never leave the modal.
5) While modal is open, screenreader needs to be trapped in the modal. Either use `aria-modal` or place `aria-hidden` on everything outside the modal.

## Possible Solution

Comply with the WAI-ARIA spec:

https://www.w3.org/TR/wai-aria-1.1/#dialog

example implementation

https://www.w3.org/TR/2017/NOTE-wai-aria-practices-1.1-20171214/examples/dialog-modal/dialog.html

## Context (Environment)

using Orbit 0.13.0

**Comment by vepor** [comment]
As we merged #958 can you @filipdanisko check if we can close this issue?

**Comment by filipdanisko** [comment]
@vepor we might but I would still rather update readme to address point 1. other than that should be good

**Comment by vepor** [comment]
Well then. Can you update it? 🙏

**Comment by filipdanisko** [comment]
@vepor yes I was about to do that :))
